runtime/precompiles/src/StakingInterface.sol

```diff
@@ -0,0 +1,85 @@
+
+    /// Check whether the specified addess is currently a collator candidate
+    function is_candidate(address collator) external view returns (bool);
+
+    // Now the dispatchables
+
+    /// Join the set of collator candidates
+    function join_candidates(uint256 amount) external;
+
+    /// Request to leave the set of candidates. If successful, the account is immediately
+    /// removed from the candidate pool to prevent selection as a collator, but unbonding is
+    /// executed with a delay of `BondDuration` rounds.
+    function leave_candidates() external;
+
+    /// Temporarily leave the set of collator candidates without unbonding
+    function go_offline() external;
+
+    /// Rejoin the set of collator candidates if previously had called `go_offline`
+    function go_online() external;
+
+    /// Bond more for collator candidates
+    function candidate_bond_more(uint256 more) external;
+
+    /// Bond less for collator candidates
+    function candidate_bond_less(uint256 less) external;
+
+    /// If caller is not a nominator, then join the set of nominators
+    /// If caller is a nominator, then makes nomination to change their nomination state
+    function nominate(address collator, uint256 amount) external;
+
+    /// Leave the set of nominators and, by implication, revoke all ongoing nominations
+    function leave_nominators() external;
+
+    /// Revoke an existing nomination
+    function revoke_nomination(address collator) external;
+
+    /// Bond more for nominators with respect to a specific collator candidate
+    function nominator_bond_more(address candidate, uint256 more) external;
+
+    /// Bond less for nominators with respect to a specific nominator candidate
+    function nominator_bond_less(address candidate, uint256 less) external;
+}
+
+// These are the selectors generated by remix following this advice
+// https://ethereum.stackexchange.com/a/73405/9963
+// Eventually we will probably want a better way of generating these and copying them to Rust
+// {
+// 	"42a72461": "at_stake(uint256,address)",
+// 	"289b6ba7": "candidate_bond_less(uint256)",
+// 	"c57bd3a8": "candidate_bond_more(uint256)",
+// 	"96b41b5b": "candidate_pool()",
+// 	"b0c0081f": "collator_commission()",
+// 	"427c3c14": "exit_queue()",
+// 	"767e0450": "go_offline()",
+// 	"d2f73ceb": "go_online()",
+// 	"10db2de9": "inflation_config()",
+// 	"8545c833": "is_candidate(address)",
+// 	"8e5080e7": "is_nominator(address)",
+// 	"8f6d27c7": "is_selected_candidate(address)",
+// 	"ad76ed5a": "join_candidates(uint256)",
+// 	"b7694219": "leave_candidates()",
+// 	"e8d68a37": "leave_nominators()",
+// 	"82f2c8df": "nominate(address,uint256)",
+// 	"f6a52569": "nominator_bond_less(address,uint256)",
+// 	"971d44c8": "nominator_bond_more(address,uint256)",
+// 	"9799b4e7": "points(uint256)",
+// 	"4b65c34b": "revoke_nomination(address)",
+// 	"98807d84": "staked(address)",
+// 	"7b46e61b": "total_selected()"
```

Comment:
There's a bunch more accessors in here that we could implement. They can be in follow-ups if you prefer, but we should at least do min_nominator stake. I thought that was include din my list at some point, but maybe not.

Wait, actually, did you remove stuff from the solidity code in this file? I had typed these all in to generate the signatures.

Reply:
> Wait, actually, did you remove stuff from the solidity code in this file? I had typed these all in to generate the signatures.

Yes, I removed all unimplemented accessors from the solidity code. I think it makes the most sense to add them back when they're implemented instead of beforehand.

> They can be in follow-ups if you prefer, but we should at least do min_nominator stake. I thought that was include din my list at some point, but maybe not.

MinNominatorStk is a constant declared in the trait Config, it is not a mutable storage item so i don't really think we need to implement an accessor for it, until/unless we make it a mutable storage item.

Reply:
Is the goal to have 1:1 wrapper functions for everything exposed from the pallet?

I'm resisting the urge to suggest we could generate this. Oops...

Reply:
> But I'll do it anyway, just so the code is easier to maintain I guess.

Nope, am stuck. Don't know how to get the associated type for trait Config set at compile time

```rust
error[E0575]: expected method or associated constant, found associated type `parachain_staking::Config::MinNomination`
   --> runtime/precompiles/src/staking.rs:251:30
    |
251 |         let min_nomination: U256 = <Runtime as parachain_staking::Config>::MinNomination.into();
    |                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: can't use a type alias as a constructor

error: aborting due to previous error
```